concordium-std/src/impls.rs

```diff
@@ -1563,7 +1606,12 @@ impl<A> UnwrapAbort for Option<A> {
     type Unwrap = A;
 
     #[inline(always)]
-    fn unwrap_abort(self) -> Self::Unwrap { self.unwrap_or_else(|| crate::trap()) }
+    fn unwrap_abort(self) -> Self::Unwrap {
+        match self {
+            Some(x) => x,
+            None => crate::trap(),
+        }
+    }
```

Comment:
Why was this change made?

Please avoid unrelated changes unless they are bugs. In which case it is still better to document a bug and fix it in a separate PR.